scripts/calculateRevenue/protocols/beefy.ts

```diff
@@ -1,11 +1,100 @@
-import { RevenueResult } from '../../types'
+import { RevenueResult, NetworkId } from '../../types'
 import { fetchWithTimeout } from '../../utils/fetchWithTimeout'
-
+import { getViemPublicClient, getStrategyContract } from './utils/viem'
+import { Address } from 'viem'
 export type BeefyVaultTvlData = [string, number]
 
 const BEEFY_API_URL = 'https://databarn.beefy.com/api/v1/beefy'
+const DEFI_LLAMA_API_URL = 'https://coins.llama.fi'
+
 const ONE_WEEK = 7 * 24 * 60 * 60 * 1000
 
+const NETWORK_ID_TO_DEFI_LLAMA_CHAIN: Partial<{
+  [networkId in NetworkId]: string // eslint-disable-line @typescript-eslint/no-unused-vars
+}> = {
+  [NetworkId['ethereum-mainnet']]: 'Ethereum',
+  [NetworkId['arbitrum-one']]: 'Arbitrum',
+  [NetworkId['op-mainnet']]: 'Optimism',
+  [NetworkId['celo-mainnet']]: 'Celo',
+  [NetworkId['polygon-pos-mainnet']]: 'Polygon',
+  [NetworkId['base-mainnet']]: 'base',
+}
+
+export interface BlockTimestampData {
+  height: number
+  timestamp: number
+}
+
+interface FeeEvent {
+  beefyFee: number | bigint
+  timestamp: Date
+}
+
+// TODO: Memoize this function so it's not repeated for every user address
+/**
+ * Uses the DefiLlama API to fetch the block number nearest a given timestamp
+ */
+export async function getNearestBlock(
+  networkId: NetworkId,
+  timestamp: Date,
+): Promise<number> {
+  const unixTimestamp = Math.floor(timestamp.getTime() / 1000)
+  const defiLlamaChain = NETWORK_ID_TO_DEFI_LLAMA_CHAIN[networkId]
+
+  const response = await fetchWithTimeout(
+    `${DEFI_LLAMA_API_URL}/block/${defiLlamaChain}/${unixTimestamp}`,
+  )
+  if (!response.ok) {
+    console.log(
+      `${DEFI_LLAMA_API_URL}/block/${defiLlamaChain}/${unixTimestamp}`,
```

Comment:
nit: pull this to a variable, so these don't accidentally diverge

Reply:
Oops, this log actually shouldn't have made it in to the PR.